### ISSUE TITLE
Steam notifications cannot be disabled

### ISSUE THREAD

**Issue by pdknsk** [ValveSoftware/steam-for-linux#3239]
I have unchecked _Enabled the Steam Overlay while in-game_ in settings, yet it still shows achievements or card messages. The option is set properly, because when I try to re-enable it for a single game I get an error stating _You have disabled Steam Community In-Game in your Steam wide settings_.

**Comment by makoConstruct** [comment]
Seconded. Acheivement popups really harm the experience more than I can communicate. Still happens for me in fullscreen. I'm running a debian system.

**Comment by Plagman** [comment]
Are you getting Steam notifications rather than overlay notifications? You can also disable Steam desktop notifications separately from the in-game setting.

**Comment by makoConstruct** [comment]
They are positioned relative to the desktop, but oddly they're interrupting the rendering of the game even when it's in windowed mode, additionally I'm not seeing any setting for desktop notifications, if that's a separate thing. Enabling in-game overlay(in windowed mode) stops the game from flashing when an achievement pops up, but it does not provide any additional avenues to disable them.

**Comment by mesilliac** [comment]
This is extremely annoying.

The assignee doesn't seem to be working on it.

Is it really that hard to disable notifications when the overlay is turned off?

Annoying, distracting, immersion-breaking achievement popups are actively preventing me from buying new games on Steam.

**Comment by harlin** [comment]
Any update on this?
There is recent bug with some Unity games, which freeze when any Steam popup appears, which is EXTREMELY annoying.
(Samples: Victor Vran, Tahira: Echoes of the Astral Empire)

**Comment by haasn** [comment]
> I was mistaken. In windowed games this is still the case.

Does this mean setting the game to fullscreen mode is a work-around? If so, good to know. Achievements are extremely fucking annoying and one of the reasons I will absolutely refuse to use the piece of shit steam client unless there's literally no alternative way to play a game.

**Comment by haasn** [comment]
> Does this mean setting the game to fullscreen mode is a work-around?

Doesn't work.

**Comment by haasn** [comment]
There is a working work-around here:

https://steamcommunity.com/discussions/forum/1/617329920710103124/

It's not really a perfect work-around though since judging by the method, it doesn't actually disable the steam overlay crap, it just makes the notifications have 0 duration. So it probably still drains resources and/or causes crashes as mentioned.

Although, that being said, maybe `chmod -x` on ~/.steam/steam/ubuntu12_{32,64}/gameoverlayrenderer.so` prevents it from successfully being LD_PRELOADed?

**Comment by tuxdelux** [comment]
"Cook, Serve, Delicious" and "Thimbleweed Park" and "The Count Lucanor" and "The Last Door 2" and "The Spatials" will also freeze on my system with a GM45 Express Intel graphics chip. Very annoying.

**Comment by RockyTV** [comment]
Why isn't it possible to disable notification popup like it is on Windows? This really annoys me because there is a bug which floods notifications about your friends that started playing a game, and when you're in game it's worse. Another workaround is to set yourself as invisible on chat and it'll stop popping up. Other notifications will still be enabled, though.

**Comment by alexzk1** [comment]
So you say, it is 5 years there ? And still cannot be fixed  ? :D 
Can you just delete that pop-up completely so it will be just absent. As it has many other drawbacks, like stealing focus from game.

**Comment by alexzk1** [comment]
Please fix it already. Tired to press RESET 5 times per day because of those pops. Some games (not all) hang if alt-tab or any pop above.

**Comment by mesilliac** [comment]
It really cannot be hard to fix this. Hell, i'll even do it. Just give me access to the code. I'll sign whatever NDA you need.

**Comment by alexzk1** [comment]
Hello ? Fix / disable / completely remove those popups whatever is faster!
Special cruelty is to get rare random drop, get achievement pop, press reset after this and never have this drop back again as game was not saved!
Now I play in mode "how to AVOID any achievements" because of those pop-ups. Those make me angry to say least.

**Comment by olehj** [comment]
I like to use Steam Overlay, but I would love to disable the popups because CS:GO is freezing for 3-4 seconds while the popup is fading in and fading out (or fix the issue itself of course).

**Comment by TheSHEEEP** [comment]
How is this not yet fixed? 
It really seems like it should be simple?

**Comment by mamoit** [comment]
Doom Eternal on proton crashes every time an achievement is earned, which makes me not want to get achievements.
Tried to disable the overlay all together, but the achievement popup still comes up.

**Comment by tomboland** [comment]
Agreed.  This is killing Doom Eternal for me.  It just took me 5-6 attempts to get through a particularly difficult encounter (a slayer gate), and I finally beat it, then boom.  achievement notification pops up, and crashes the game.  I must say though, this isn't Steam's bug, it's a but in the game/proton/nvidia driver or something, because if I don't kill my email client before starting the game, a desktop notification will kill the game in exactly the same way.  I would  still _love_ to be able to switch off the notifications though.  Please! :)

**Comment by frozenMustelid** [comment]
If I had known that it was possible to submit bug reports to Valve, I would have reported this years ago. I hate, hate, HATE this bug and it's one of the primary reasons I buy from GOG whenever the game is available for both platforms. It's worth it to deal with WINE the old-fashioned way instead of the "just works" way Proton uses because I hate the constant flood of achievement popups, inventory popups, card popups, and other popups so, so much.

This bug has existed on Mint 17.x (based on Ubuntu 14.04 LTS), 18.x (Ubuntu 18.04 LTS), 20.x (Ubuntu 20.04 LTS), and probably exists on other releases but those are the only ones I've tried.

**Comment by blooalien** [comment]
Everything mentioned in this bug report thread, **plus** on **my** system, when I get an achievement popup, it gets stuck on my screen, often overlaying some important bit of the game interface, making the game unplayable until I quit and restart the game in question.  This happens to me with **every** game at some point sooner or later (usually sooner rather than later), and I can find **no possible way** to disable these achievement notifications no matter what I've tried (and I've been trying for **literal years**).  Even the "hacks" involving a custom Steam skin have not helped anything.  I agree with those here who question; "Is it **really** so hard to just allow us to disable these notifications?"  Why was a (properly working) checkbox not added to the Steam settings for this **ages ago**?  This is a (every)game-breaking **bug** and **should** have **already** been fixed, no?

**Comment by Sha1rath** [comment]
@blooalien There is a workaround: Create the steam skin to prevent notifications from being displayed. Worked for me with Doom Eternal (and I've never seen a single notification ever since).
https://steamcommunity.com/discussions/forum/1/617329920710103124/

I agree, this bug should be fixed. Maybe for it's 7th birthday?

**Comment by zlice** [comment]
Seriously, why isn't this a config item yet?

**Comment by mesilliac** [comment]
It is completely absurd that this is still a thing.

Even with the custom steam skin workaround, it still creates achievement popups for a single frame, causing stuttering and flickering.

**Comment by realnc** [comment]
Can *someone* please fix this? I just tried to play a game on Linux but once the annoying "achievement unlocked" notification popped up (taking me completely out of the moment), I booted right back to Windows.

How difficult can it be to fix this in 8 years? How much time do you need? 100 years?

**Comment by ItsRogueRen** [comment]
Are you kidding me? 8 YEARS and no one has said anything about this? Is turning off all popups just never going to be an option?

**Comment by zlice** [comment]
@slouken I'm guessing this is buried? Any way you can reassign or follow up?

**Comment by ItsRogueRen** [comment]
I've made a new issue since this seems to be all but forgotten

https://github.com/ValveSoftware/steam-for-linux/issues/9477

**Comment by zlice** [comment]
So they do know, they just don't care. Thanks.

**Comment by frozenMustelid** [comment]
So opening a duplicate issue is important enough to get a same-day response and a finger wag from one of Valve's employees, but the issue itself is still unresolved 9 years, 1 month, and 4 days later? With seemingly no progress ever having been made and no work ever having been done?

**Comment by mamoit** [comment]
Ticket triaging is not the same thing as implementing a feature or fixing a bug.
Please vote with a 👍 on the issue.
Usually issues with a lot of impact get fixed first, this only has 8 👍 so it is not that prevalent even if it impacts you.

**Comment by zlice** [comment]
**usually**. Beta recently updated a few broken config items (like hardware acceleration) that had a bug open for several years. The fact there are games only Steam has that are affected by this is lost money to the monopoly.

**Comment by mesilliac** [comment]
Well after ten years of this being an unresolved problem that would have taken maybe an hour of development time to fix (yes i'm unimpressed, and yes they've lost more money than that would cost just to me not buying a few steam games i would otherwise have bought), it's finally resolved as a side-effect of an overhauled notification system.

At least with the latest client beta, there's a new "notifications" tab in settings where you can turn all notifications, including achievement popups, off.

**Comment by ItsRogueRen** [comment]
Been using the new client for the last few weeks and have been able to have all notifications, including achievement popups, disabled while playing games. FANTASTIC! No more using 3rd party tools to spoof achievements so that I don't get that annoying popup while playing my games, I just turn them off in the settings and be done with it. 

It took quite some time but I am so happy I can finally play my PC games and not be interrupted by annoying notifications I do not care about.

**Comment by kisak-valve** [comment]
Closing per the last couple comments.
